### PR TITLE
Up 3413 jsr 168 and 286 css for responder

### PR DESCRIPTION
**Pull request by MichaelVose2**
[html.docx](https://github.com/Jasig/uPortal/files/452242/html.docx)
Associated with UP-3413.

This PR addresses issues raised in UP-3413 with regards to certian jsr-168 and jsr 286 portlet classes not being supported.

The dom is outlined in the above .doc.

(nb. rebasing did not work properly, hence 8 commits which is annoying..)

### CODE REVIEW

File: uportal-war/src/main/webapp/media/skins/respondr/common/less/jsr-168.less [hunk 1]
```diff
@@ -0,0 +1,311 @@
+///@import "../bootstrap/variables.less";
```
Comment (ChristianMurphy) [inline uportal-war/src/main/webapp/media/skins/respondr/common/less/jsr-168.less]: remove commented out import

**Comment by ChristianMurphy** [comment]
@apetro @drewwills does this code already exist somewhere in uPortal?
Github search and `grep` aren't turning up anything for me.
However there is a pretty detailed page on [JSR-168 css](https://wiki.jasig.org/display/UPC/JSR-168+PLT.C+CSS+Style+Definitions) in the wiki, that seems to imply this has been done before.

**Comment by MichaelVose2** [comment]
Hello.

The document you are linking to seems to be a copy and paste of oracles guidance which is as follows..https://java.net/downloads/portletspec3/WorkingDocs/PortletSpec3-20130708.pdf . UI is metioned on page 231 to page 236.

Even if JSR 168 CSS does exist, it would not be utilising LESS and thus the end result would be inconsistant colouring as existing defined user variables should be used.

I am currently in Italy however I can do this when I get back in 2 weeks.

Do the community have any ideas as to how to remove the 8 commits and squash them into 1 as I tried, but clearly, did it incorrectly.

I was unaware of styleguide at the time.

Please be aware that I expect people to use html, css, and js within the bootstrap framework in order to do menus etc. as the aim of this work was to not reinvent the wheel, but to support older portlets.

**Comment by MichaelVose2** [comment]
Hello.

The document you are linking to seems to be a copy and paste of oracles guidance which is as follows..https://java.net/downloads/portletspec3/WorkingDocs/PortletSpec3-20130708.pdf . UI is metioned on page 231 to page 236.

Even if JSR 168 CSS does exist, it would not be utilising LESS and thus the end result would be inconsistant colouring as existing defined user variables should be used.

I am currently in Italy however I can implement any desired changed in around 2 weeks.

Do the community have any ideas as to how to remove the 8 commits and squash them into 1 as I tried, but clearly, did it incorrectly.

I was unaware of styleguide at the time.

Please be aware that I expect people to use html, css, and js within the bootstrap framework in order to do menus etc. as the aim of this work was to not reinvent the wheel, but to support older portlets.

________________________________

**Comment by ChristianMurphy** [comment]
> Do the community have any ideas as to how to remove the 8 commits and squash them into 1 as I tried, but clearly, did it incorrectly.

Before doing any form of history rewriting I would suggest taking a backup of the changes. :lock: 

then these commands should help :+1: 

``` shell
# bring code inline with upstream uPortal
git merge https://github.com/Jasig/uPortal master --rebase

# interactively squash past 8 commits
git rebase --interactive HEAD~8
```

source: https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

**Comment by MichaelVose2** [comment]
Hello Christian,

I have made the LESS changes here:

https://github.com/Jasig/uPortal/pull/711

and I am about to attempt to squelch after taking a backup, however, with regards to indentation,  can you confirm if there are any specific guidelines for less within the community, especially with regards to nested LESS – ie.

.hello{
    .world{}
}

As I am unsure if it is better to indent selectors in the manner in which paramaters are indented such as:


.hello{
    Color:#fff;
    .world{
       Color:#000;
    }
}

Or should each nested class be indented in it’s own right such as:

.hello{
    Color:#fff;
        .world{
           Color:#000;
       }
}

Kindly,
Michael Vose.

**Comment by ChristianMurphy** [comment]
>  can you confirm if there are any specific guidelines for less within the community, especially with regards to nested LESS

Around nested less the most common style is

``` less
.selector {
    // Rules are indented one level within a selector block
    color: blue;

    // Nested selectors come after rules and are at the same indent level as sibling rules
    .nested-selector {
        color: red;

        .nested-nested-selector {}
    }
}
```

**Comment by ChristianMurphy** [comment]
@MichaelVose2 has this been superseded by #770 ?

**Comment by MichaelVose2** [comment]
Yes, however, it fixes the issues outlined for the above uPortal issue within the responder skin. I dont think it is possible to rework this to work within "common" without a lot of effort, as clearly the colour scheme would need to alter per skin.
________________________________

**Comment by drewwills** [comment]
The subsequent PR from @MichaelVose2 was applied, so closing this one.

**Pull request closed without merging**